cmd/metrics.go

```diff
@@ -0,0 +1,173 @@
+		},
+		[]string{"project", "topics", "ref", "stage", "job_name"},
+	)
+
+	lastRunJobStatus = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "gitlab_ci_pipeline_last_job_run_status",
+			Help: "Status of the most recent job",
+		},
+		[]string{"project", "topics", "ref", "stage", "job_name", "status"},
+	)
+
+	lastRunJobArtifactSize = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "gitlab_ci_pipeline_last_job_run_artifact_size",
+			Help: "Filesize of the most recent job artifacts",
+		},
+		[]string{"project", "topics", "ref", "stage", "job_name"},
+	)
+
+	timeSinceLastJobRun = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "gitlab_ci_pipeline_time_since_last_job_run_seconds",
+			Help: "Elapsed time since most recent GitLab CI job run.",
+		},
+		[]string{"project", "topics", "ref", "stage", "job_name"},
+	)
+
+	jobRunCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "gitlab_ci_pipeline_job_run_count",
+			Help: "GitLab CI pipeline job run count",
+		},
+		[]string{"project", "topics", "ref", "stage", "job_name"},
+	)
+
+	lastRunID = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "gitlab_ci_pipeline_last_run_id",
+			Help: "ID of the most recent pipeline",
+		},
+		[]string{"project", "topics", "ref"},
+	)
+
+	lastRunStatus = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "gitlab_ci_pipeline_last_run_status",
+			Help: "Status of the most recent pipeline",
+		},
+		[]string{"project", "topics", "ref", "status"},
+	)
+
+	runCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "gitlab_ci_pipeline_run_count",
+			Help: "GitLab CI pipeline run count",
+		},
+		[]string{"project", "topics", "ref"},
+	)
+
+	timeSinceLastRun = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "gitlab_ci_pipeline_time_since_last_run_seconds",
+			Help: "Elapsed time since most recent GitLab CI pipeline run.",
+		},
+		[]string{"project", "topics", "ref"},
+	)
+
+	pipelineVariables = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "gitlab_ci_pipeline_run_count_with_variable",
+			Help: "Count of pipelines with variables",
+		},
+		[]string{"project", "ref", "pipeline_variables"},
+	)
+)
+
+var (
+	registry       = prometheus.NewRegistry()
+	defaultMetrics = []prometheus.Collector{
+		coverage,
+		lastRunDuration,
+		lastRunJobDuration,
+		lastRunJobStatus,
+		lastRunJobArtifactSize,
+		timeSinceLastJobRun,
+		jobRunCount,
+		lastRunID,
+		lastRunStatus,
+		runCount,
+		timeSinceLastRun,
+		pipelineVariables,
+	}
+)
+
+func registerMetricOn(registry *prometheus.Registry, metrics ...prometheus.Collector) {
+	for _, m := range metrics {
+		if err := registry.Register(m); err != nil {
+			panic(fmt.Errorf("could not add provided metric '%v' to the Prometheus registry: %v", m, err))
+		}
+	}
+}
+
+func emitStatusMetric(metric *prometheus.GaugeVec, labelValuess []string, statuses []string, status string, sparseMetrics bool) {
```

Comment:
is `labelValuess` a misspell or on purpose ? 🤔

Reply:
nope, thanks 👍